katz/src/main/kotlin/katz/NonEmptyList.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2017 The Katz Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package katz
+
+/**
+ * A List that can not be empty
+ */
+class NonEmptyList<out A> private constructor(
+        override val head: A,
+        override val tail: List<A>,
+        val all: List<A>) : NonEmptyCollection<A> {
```

Comment:
I think we should remove all references to being part of the  colections interface since we want to encourage proper APIs and not those broken java ones.